lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,431 @@
+            # If the number is not self-managed, look for key words
+            #if not two_way_record.get('self_managed'):
+            #    logger.info('Service is not self-managed')
+            #    keyword_phrase = detected_keyword(inbound_sms.get('messageBody', ''))
+            #    if keyword_phrase:
+            #        send_message(two_way_record.get('originationNumber', ''),
+            #                     two_way_record.get('destinationNumber', ''),
+            #                     keyword_phrase)
+
+            # Forward inbound_sms to associated service
+            logger.info(
+                f'Forwarding inbound SMS to service: {two_way_record.get("service_id")} . UrlEndpoint: {two_way_record.get("url_endpoint")}')
+            
+            result_of_forwarding = forward_to_service(inbound_sms, two_way_record.get('url_endpoint', ''))
+
+            if not result_of_forwarding:
+                logger.info('failed to make request.  Placing request back on retry')
+                # return 400 to have message put back on feeder queue with a visibility timeout to delay re-processing
+                return create_response(400)
+
+        except KeyError as e:
+            logger.critical(f'Unable to find two_way_record for: {inbound_sms.get("destinationNumber")}')
+            logger.exception(e)
+            # Deadletter
+            push_to_sqs(event_body, False)
+        except Exception as e:
+            logger.critical(f'Unhandled exception in handler')
+            logger.exception(e)
+            # Deadletter
+            push_to_sqs(event_body, False)
+
+    # return 200 to have message removed from feeder queue
+    return create_response(200)
+
+def create_response(status_code: int):
+    """
+    Create response object to return after lambda completes processing.
+    """
+    response = {
+        "statusCode": status_code,
+        "isBase64Encoded": False,
+        "body": "Success" if status_code == 200 else "Failure"
+    }
+
+    return response
+
+def valid_event_body(event_data: dict) -> bool:
+    """
+    Verify that the event body's message that comes in, contains the following 3 keys.  
+    If any are missing then processing should fail
+    """
+    if event_data.get('destinationNumber') is None:
+        return False
+    if event_data.get('originationNumber') is None:
+        return False
+    if event_data.get('messageBody') is None:
+        return False
+
+    return True
+
+def valid_event(event_data: dict) -> bool:
+    """
+    Ensure the event has all the necessary fields
+    """
+    try:
+        if 'Records' in event_data:
+            for record in event_data.get('Records'):
+                if record.get('body') is None:
+                    # Log specific record causing issues
+                    logger.critical(f'Failed to retrieve body in record: {record}')
+                    return False
+        return True
```

Comment:
If there is no 'Records' field in the event_data then it is an invalid event.  Then for each record in records, need to verify that there is a "body" element otherwise it's invalid.